dali/python/nvidia/dali/external_source.py

```diff
@@ -94,6 +99,23 @@ def reset_indices(self):
         self.current_iter = 0
         self.current_sample = 0
 
+    def schedule_batch(self, pool, context_i, batch_size):
+        pool.schedule_batch(context_i, self.current_iter, [
+            self.callback_args(i) for i in range(batch_size)
+        ])
+        self.current_sample += batch_size
+        self.current_iter += 1
+
+    def feed_and_schedule(self, pipeline, pool, context_i, batch_size):
+        try:
+            callback_out = pool.receive_batch(context_i)
+            self.feed(pipeline, callback_out, batch_size)
+            self.schedule_batch(pool, context_i, batch_size)
```

Comment:
Maybe schedule first and then feed as feed could last longer than scheduling.

Reply:
But you need to handle StopIteration properly and remember to feed the last batch anyway.

Reply:
StopIteration is raised by receive_batch not by schedule_batch, so the order between feed and schedule should not matter at all.

Reply:
ok

Reply:
changed the order